packages/yourloops/test/components/dashboard-widgets/medical-files/medical-record-list.test.tsx

```diff
@@ -0,0 +1,206 @@
+import { act, fireEvent, render, screen, waitFor, within } from "@testing-library/react";
+import { buildTeam, buildTeamMember, createPatient } from "../../../common/utils";
+import { CategoryProps } from "../../../../components/dashboard-widgets/medical-files/medical-files-widget";
+import MedicalRecordList from "../../../../components/dashboard-widgets/medical-files/medical-record-list";
+import * as authHookMock from "../../../../lib/auth";
+import User from "../../../../lib/auth/user";
+import MedicalFilesApi from "../../../../lib/medical-files/medical-files-api";
+import { MedicalRecord } from "../../../../lib/medical-files/model";
+import { MedicalRecordEditDialogProps } from "../../../../components/dialogs/medical-record-edit-dialog";
+import { MedicalRecordDeleteDialogProps } from "../../../../components/dialogs/medical-record-delete-dialog";
+
+/* eslint-disable react/display-name */
+jest.mock("../../../../lib/auth");
+jest.mock("../../../../components/dialogs/medical-record-edit-dialog", () => (props: MedicalRecordEditDialogProps) => {
+  return (
+    <div aria-label="mock-edit-dialog">
+      <button onClick={() => props.onSaved({} as MedicalRecord)}>mock-save-button</button>
+      <button onClick={() => props.onClose()}>mock-cancel-button</button>
+    </div>
+  );
+});
+jest.mock("../../../../components/dialogs/medical-record-delete-dialog", () => (props: MedicalRecordDeleteDialogProps) => {
+  return (
+    <div aria-label="mock-delete-dialog">
+      <button onClick={() => props.onDelete("fakeId")}>mock-delete-button</button>
+      <button onClick={() => props.onClose()}>mock-cancel-button</button>
+    </div>
+  );
+});
+
+describe("Medical Records", () => {
+  const patient = createPatient("fakePatientId", []);
+  const adminMember = buildTeamMember();
+  const patientMember = buildTeamMember("fakeTeamId", patient.userid);
+  const remoteMonitoringTeam = buildTeam("fakeTeamId", [adminMember, patientMember]);
+  const medicalRecords: MedicalRecord[] = [
+    {
+      id: "fakeId",
+      authorId: "fakeAuthorId",
+      creationDate: "2022-05-23",
+      patientId: "PatientId",
+      teamId: "teamId",
+      diagnosis: "diag1",
+      progressionProposal: "proposal1",
+      trainingSubject: "training1",
+    },
+    {
+      id: "fakeId2",
+      authorId: "fakeAuthorId2",
+      creationDate: "2022-05-24",
+      patientId: "PatientId2",
+      teamId: "teamId2",
+      diagnosis: "diag2",
+      progressionProposal: "proposal2",
+      trainingSubject: "training2",
+    },
+  ];
+
+  const getMedicalRecordsSpy = () => {
+    return jest.spyOn(MedicalFilesApi, "getMedicalRecords").mockResolvedValue([...medicalRecords]);
+  };
+
+  function getMedicalRecordListJSX(props: CategoryProps = {
+    patientId: patient.userid,
+    teamId: remoteMonitoringTeam.id,
+  }): JSX.Element {
+    return <MedicalRecordList {...props} />;
+  }
+
+  async function renderComponent() {
+    render(getMedicalRecordListJSX());
```

Comment:
Should be changed to: 


```
    await act(async () => {
      render(getMedicalRecordListJSX());
      await waitFor(() => expect(getMedicalRecordsSpy()).toHaveBeenCalled());
    });
```
That way all the tests no longer need to have an `act` expect the first one that will work but show a warning in the console.

To fix the issue, add the function:

```
async function renderComponentForEmptyMedicalRecordsList() {
    render(getMedicalRecordListJSX());
    await waitFor(() => expect(getMedicalRecordsSpy()).toHaveBeenCalled());
  }
```

In the first test, call this function instead of `renderComponent`. This should fix the warning.